File: java/src/main/java/io/fivexer/sdk/model/LearningStats.java
package io.fivexer.sdk.model;

/**
 * Aggregate reward statistics.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class LearningStats {
    private int decisions;
    private int rewards;
    private double totalReward;
    private double averageReward;

    public int getDecisions() { return decisions; }
    public int getRewards() { return rewards; }
    public double getTotalReward() { return totalReward; }
    public double getAverageReward() { return averageReward; }
}

File: php/src/Model/LearningStats.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Aggregate reward statistics.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class LearningStats
{
    public function __construct(
        public readonly int $decisions,
        public readonly int $rewards,
        public readonly float $totalReward,
        public readonly float $averageReward,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            decisions: (int) ($data['decisions'] ?? 0),
            rewards: (int) ($data['rewards'] ?? 0),
            totalReward: (float) ($data['totalReward'] ?? 0),
            averageReward: (float) ($data['averageReward'] ?? 0),
        );
    }
}
